pkg/apis/application/v1alpha1/types.go

```diff
@@ -669,10 +673,11 @@ func (r ResourceResults) PruningRequired() (num int) {
 
 // RevisionHistory contains information relevant to an application deployment
 type RevisionHistory struct {
-	Revision   string            `json:"revision" protobuf:"bytes,2,opt,name=revision"`
-	DeployedAt metav1.Time       `json:"deployedAt" protobuf:"bytes,4,opt,name=deployedAt"`
-	ID         int64             `json:"id" protobuf:"bytes,5,opt,name=id"`
-	Source     ApplicationSource `json:"source,omitempty" protobuf:"bytes,6,opt,name=source"`
+	Revision        string            `json:"revision" protobuf:"bytes,2,opt,name=revision"`
+	DeployedAt      metav1.Time       `json:"deployedAt" protobuf:"bytes,4,opt,name=deployedAt"`
+	ID              int64             `json:"id" protobuf:"bytes,5,opt,name=id"`
+	Source          ApplicationSource `json:"source,omitempty" protobuf:"bytes,6,opt,name=source"`
+	DeployStartedAt metav1.Time       `json:"deployStartedAt,omitempty" protobuf:"bytes,7,opt,name=deployStartedAt"`
 }
```

Comment:
I know the members of this type have not been documented before, but while you're at that location already, would you mind adding comments to them? The comments will be used throughout the documentation, and also end up as field descriptions in the CRD. I think this would be good :)

Reply:
done 👍